src/url-mapper.js

```diff
@@ -63,12 +65,7 @@ export default class UrlMapper {
   set(url, newUrl, isLocal, isActive = true) {
     isLocal = !!isLocal;
 
-    // fix the url, because hoxy will add a slash to urls without path..
-    if (!isLocal && newUrl.split('/').length === 3 && newUrl.indexOf('?') === -1) {
-      newUrl += '/';
-    }
-
-    url = url.trim();
+    url = (url.endsWith('/') ? url.substring(0, url.length - 1) : url).trim();
```

Comment:
Could you get rid of the code duplication here?
Maybe we can move that to a helper which we can properly test.